FlowCrypt/src/main/java/com/flowcrypt/email/util/GeneralUtil.kt

```diff
@@ -89,17 +89,22 @@ class GeneralUtil {
 
     /**
      * Checking for an Internet connection.
-     * See https://developer.android.com/training/monitoring-device-state/connectivity-monitoring#DetermineConnection.
-     * Because [NetworkInfo.isConnectedOrConnecting] is deprecated we will use [NetworkInfo.isConnected]
      *
      * @param context Interface to global information about an application environment.
-     * @return <tt>boolean</tt> true - a connection available, false if otherwise.
+     * @return true - a connection available, false if otherwise.
      */
-    //todo-denbond7 Need to fix deprecation
     fun isConnected(context: Context?): Boolean {
-      val cm = context?.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
-      val activeNetwork = cm?.activeNetworkInfo
-      return activeNetwork?.isConnected ?: false
+      val connectivityManager =
+        context?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+      val network = connectivityManager.activeNetwork ?: return false
+      val networkCapabilities = connectivityManager.getNetworkCapabilities(network) ?: return false
+      return when {
```

Comment:
can be just `return networkCapabilities ... ` (whole condition which results `true`)